core/storageproviders/s3Storage.go

```diff
@@ -85,6 +88,8 @@ func (s *S3Storage) GenerateRemotePlaylist(playlist string, variant models.Varia
 			fullRemotePath := variant.GetSegmentForFilename(line)
 			if fullRemotePath == nil {
 				line = ""
+			} else if s.s3ServingEndpoint != "" {
+				line = fmt.Sprintf("%s/%s/%s", s.s3ServingEndpoint, config.Config.PrivateHLSPath, fullRemotePath.RelativeUploadPath)
```

Comment:
Could you use `config.Config.GetPrivateHLSSavePath()` instead of `config.Config.PrivateHLSPath`?  It takes into account defaults if people don't have it specified in their config.